src/main/java/moe/banana/jsonapi2/HasOne.java

```diff
@@ -19,6 +19,19 @@ public T get() throws ResourceNotFoundException {
         return (T) resource._doc.find(linkage);
     }
 
+    public T get(T defaultValue) {
+        try {
+            return get();
+        }
+        catch (ResourceNotFoundException e){
+            return defaultValue;
+        }
+    }
+
+    public boolean resourceExists(){
+        return resource._doc.contains(linkage);
+    }
+
```

Comment:
Yeah, I wasn't to happy with the name either!

Perhaps we could reverse it and go with isEmpty() the same as collections
and strings?  That would work for HasMany?

On Wed, 7 Sep 2016, 6:24 PM 羽喵 <notifications@github.com> wrote:

> In src/main/java/moe/banana/jsonapi2/HasOne.java
> <https://github.com/kamikat/moshi-jsonapi/pull/14#discussion_r77778879>:
>
> > @@ -19,6 +19,19 @@ public T get() throws ResourceNotFoundException {
> >          return (T) resource._doc.find(linkage);
> >      }
> >
> > +    public T get(T defaultValue) {
> > +        try {
> > +            return get();
> > +        }
> > +        catch (ResourceNotFoundException e){
> > +            return defaultValue;
> > +        }
> > +    }
> > +
> > +    public boolean resourceExists(){
> > +        return resource._doc.contains(linkage);
> > +    }
> > +
>
> The name resourceExists is considered too verbose, exists or isAccessible
> can be better to read/write. Maybe we should rename the function with a
> more generally acceptable name so that HasMany can implement the same
> semantic with the same member function. isResolvable is one of the names
> I can come up with.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kamikat/moshi-jsonapi/pull/14/files/7167b318eac945758255aec3d983d0cfdbaea324..dbb1355e0ff445f02cacb6168cdaff999a37b34e#r77778879>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA3EbIC7dv3ExVfUyBeO-wgm_8X9vLy0ks5qnnTIgaJpZM4J2lGB>
> .
>